### ISSUE TITLE
Idea for browser support

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/super-regex#8]
It might be possible to create and spawn a temporary web worker from a datauri, execute the regex within and then [terminate](https://developer.mozilla.org/en-US/docs/Web/API/Worker/terminate) it on timeout.

(Not really interested in implementing this, but it may be a nice exercise for someone 😉)

**Comment by sindresorhus** [comment]
Yeah, that could work. There are ways to make a web worker block the main process by using a shared array buffer.

**Comment by silverwind** [comment]
Yeah, I guess the main challenge will not be to spawn the worker, but keep it all in synchronous code, so that the same API can be provided.

**Comment by sindresorhus** [comment]
I have added async methods now. I'm not going to bother with synchronous methods for the browser.
